[masonry] getComputedStyle should return masonry

stwrt opened the pull request:
#### c61b4bd8c681b8982c2c46af366f6654ac162e2d
<pre>
[masonry] getComputedStyle should return masonry
<a href="https://bugs.webkit.org/show_bug.cgi?id=279387">https://bugs.webkit.org/show_bug.cgi?id=279387</a>
<a href="https://rdar.apple.com/problem/135602712">rdar://problem/135602712</a>

Reviewed by Sammy Gill.

getComputedStyle() should return a value of masonry when computing grid-template-columns or grid-template-rows.
Returning a px value as we were doing before in the test case is not correct.

<a href="https://drafts.csswg.org/css-grid-3/#grid-template-masonry">https://drafts.csswg.org/css-grid-3/#grid-template-masonry</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/masonry-grid-template-columns-computed-withcontent-expected.txt:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::valueForGridTrackList):
(WebCore::rendererCanHaveTrimmedMargin):
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):

Canonical link: <a href="https://commits.webkit.org/283429@main">https://commits.webkit.org/283429@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/76339d042788c01fb088b4b8c907d3b717e912f2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66139 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45512 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18758 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70171 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16749 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/68257 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53311 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17030 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53076 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11660 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69206 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41985 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57258 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33712 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38656 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/14636 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15625 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60551 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14982 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71873 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10094 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14389 "Found 1 new test failure: http/wpt/mediastream/transfer-mediastreamtrack-to-worker.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60397 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10126 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57320 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60688 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14608 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8335 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1970 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41320 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42396 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43579 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42140 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.cpp:
```diff
@@ -1194,10 +1194,18 @@ static Ref<CSSValue> valueForGridTrackList(GridTrackSizingDirection direction, R
+    // If masonry is specified for both grid-template-columns and grid-template-rows,
+    // then the used value for grid-template-columns is 'none'.
+    if (style.gridMasonryRows() && style.gridMasonryColumns())
+        return direction == GridTrackSizingDirection::ForColumns ? CSSPrimitiveValue::create(CSSValueNone) : CSSPrimitiveValue::create(CSSValueMasonry);
```
This is just changing the computed value, not the used value

stwrt replied:
I misread the spec here. Fixed the code now.

stwrt committed:
[masonry] getComputedStyle should return masonry
https://bugs.webkit.org/show_bug.cgi?id=279387
rdar://problem/135602712

Reviewed by Sammy Gill.

getComputedStyle() should return a value of masonry when computing grid-template-columns or grid-template-rows.
Returning a px value as we were doing before in the test case is not correct.

https://drafts.csswg.org/css-grid-3/#grid-template-masonry

* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/masonry-grid-template-columns-computed-withcontent-expected.txt:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::valueForGridTrackList):
(WebCore::rendererCanHaveTrimmedMargin):
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):

Canonical link: https://commits.webkit.org/283429@main
